src/components/notfound/NotFound.test.tsx

```diff
@@ -0,0 +1,34 @@
+import { fireEvent, render, screen } from "@testing-library/react";
+import * as React from "react";
+import NotFound from "./NotFound";
+import { MemoryRouter } from "react-router";
+import { Router } from "react-router-dom";
+import { createMemoryHistory } from "history";
+
+describe("NotFound component", () => {
+  it("should render NotFound component", async () => {
+    const { getByTestId, getByText } = render(
+      <MemoryRouter>
+        <NotFound />
+      </MemoryRouter>
+    );
```

Comment:
I think we can render NotFound component without using MemoryRoute, you just need to mock useHistory with something like this.

Reply:
Also the test cases doesn't have to async, I might be wrong but please double check.

Reply:
@RohitKandimalla Thanks so much for the comments. You are right those are not needed, so I removed them and just pushed in the code.

Reply:
@RohitKandimalla The new code with mock push was checked in, thanks again Rohit!!!